test/relations.test.js

```diff
@@ -3145,6 +3145,169 @@ describe('relations', function () {
 
   });
 
+  describe('hasOne with primaryKey different from model PK', function () {
+    var CompanyBoard, Boss;
+    var companyBoardId, bossId;
+
+    before(function () {
+      db = getSchema();
+      CompanyBoard = db.define('CompanyBoard', {membersNumber: Number, companyId: String});
+      Boss = db.define('Boss', {boardMembersNumber: Number, companyId: String});
+    });
+
+    it('relation can be declared with primaryKey', function () {
+      CompanyBoard.hasOne(Boss, {
+        properties: {membersNumber: 'boardMembersNumber'},
+        primaryKey: 'companyId',
+        foreignKey: 'companyId'
+      });
+      Object.keys((new Boss()).toObject()).should.containEql('companyId');
+      (new CompanyBoard()).boss.should.be.an.instanceOf(Function);
+    });
+
+    it('can be used to query data', function (done) {
+      db.automigrate(function () {
```

Comment:
Use `db.automiagrate(['CompanyBoard', 'Boss'], function() {` instead for much better performance of testing.